src/components/collection/schema/Editor/index.tsx

```diff
@@ -74,14 +78,23 @@ function CollectionSchemaEditor({ entityName, localZustandScope }: Props) {
         setCollectionData,
     ]);
 
+    // If the schema is updated via the scheme inference
+    //  of CLI button we want to fire mutate and make sure we get the latest
     useUpdateEffect(() => {
-        // If the schema is updated via the scheme inference
-        //  of CLI button we want to fire mutate and make sure we get the latest
         if (mutate && schemaUpdated) {
             void mutate();
         }
     }, [schemaUpdated]);
 
+    // If the collection was initialized we need to fire mutate so that the collection
+    //  spec editor can refresh itself and know about the new draft we just created
+    //  for the collection
+    useUpdateEffect(() => {
+        if (mutate && collectionInitializationDone) {
+            void mutate();
+        }
+    }, [mutate, collectionInitializationDone]);
```

Comment:
This allows the proper collection schema to be displayed when the `Collection` tab is open and the binding is changed.